clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -422,6 +422,55 @@ template <typename T> constexpr uint asuint(T F) {
   return __detail::bit_cast<uint, T>(F);
 }
 
+//===----------------------------------------------------------------------===//
+// asuint splitdouble builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn void asuint(double D, out uint lowbits, out int highbits)
+/// \brief Split and interprets the lowbits and highbits of double D into uints.
+/// \param D The input double.
+/// \param lowbits The output lowbits of D.
+/// \param highbits The highbits lowbits D.
+#if __is_target_arch(dxil)
```

Comment:
I might be confused here so I apologize ahead of time.  Isn't `__is_target_arch` a compile time check? I'm confused with how this work when we can change the compiler target after compilation via `-triple dxil-pc-shadermodel6.3-library` or  `-triple spirv--vulkan-library`?

Reply:
This header is included by clang when compiling HLSL sources, not when building Clang.

Reply:
So my theory was that something about  using `#elif __is_target_arch(spirv)`

is forcing us to have to specify `-cc1 -triple ` in `clang/test/Driver/hlsl-lang-targets-spirv.hlsl` instead of `-target `.

I don't fully understand what though.

Reply:
From @joaosaffran's comment on that test (https://github.com/llvm/llvm-project/pull/109331#discussion_r1779366157), it seems like the SPIR-V implementation fails to codegen through the SPIR-V backend. It has nothing to do with the preprocessor. The `clang` commands complete the codegen pipeline unless an error is encountered.

There are two concerns that brings up to me:
1) Why is an unused function being generated?
2) Why is it failing?